be/src/common/configbase.cpp

```diff
@@ -189,7 +189,23 @@ bool Field::set_value(std::string value) {
         return false;
     }
     StripWhiteSpace(&value);
-    return parse_value(value);
+    bool flag = parse_value(value);
+    if (flag) {
+        _current_set_val = value;
+    }
+    return flag;
+}
+
+void Field::set_last_update_value() {
+    _last_set_val = _current_set_val;
+}
+
+bool Field::rollback_last_value() {
+    bool flag = parse_value(_last_set_val);
+    if (flag) {
+        _current_set_val = _last_set_val;
```

Comment:
need to reset _last_set_val, `_last_set_val.clear()`.
And when doing the rollback, need to check if _last_set_val is empty or not, so the rollback can't be performed recursively.